anchors/dna/anchors/anchors.js

```diff
@@ -1,528 +1,99 @@
-    var entryType = parms === undefined ? undefined : parms.entryType == undefined ? ANCHOR_GENERIC : parms.entryType;
-    var preserveOldValueEntry = parms === undefined ? undefined : parms.preserveOldValueEntry == undefined ? false : parms.preserveOldValueEntry;
-
-    if (anchorHash === undefined) // didn't get an anchorHash
-    {
-        if (anchor === undefined)
-            throw new errorObject("Must pass either anchor or anchorHash!");
-
-        anchorHash = makeHash("anchor_base", anchor);
-    }
-    else if (anchor !== undefined) // got an anchor hash and an anchor
-        throw new errorObject("Can't pass both anchor and anchorHash!");
-
-    if (valueHash === undefined) {
-        if (value === undefined) 
-            throw new errorObject("Must pass either value or valueHash!");
-
-        valueHash = makeHash(entryType, makeAnchorValue(entryType, value));
-    }
-    else if (value !== undefined)
-        throw new errorObject("Can't pass both value and valueHash!");
-
-    // get the links on that base
-    try { var links = getLinks(anchorHash, "", { Load: true }); }
-
-    catch (err) // failed
-    {
-        if (err.holochainMessage == "hash not found")
-            throw new errorObject("Anchor does not exist!");
-        else if (err.holochainMessage == "No links for ") // no links on that base!
-            return 0;
-        else
-            throw err;
+function validateCommit(entry_type,entry,header,pkg,sources) {
+  // debug('Anchors validatePut:' + sources)
+    if (entry_type == 'anchor') {
+        return true;
     }
-
-    var count = 0;
-
-    for (var x = 0; x < links.length; x++) {
-
-        var et = links[x].Tag.split(ANCHOR_DELIMETER)[0];
-        if (links[x].Hash == valueHash && entryType == et) {
-            
-            commit("anchor_link", { Links: [{ Base: anchorHash, Link: links[x].Hash, Tag: links[x].Tag, LinkAction: HC.LinkAction.Del }] })
-    
-            // if (!preserveOldValueEntry && count == 0)
-            //     remove(links[x].Hash, "anchors:removeFromList");
-
-            count++;
-        }
+    if (entry_type == 'anchor_link') {
+        return true;
     }
-
-    return count;
-
-}
-
-// a version of makeHash that uses anchorGeneric when no entryType is specified
-function makeAnchorValueHash(parms)
-{
-    return makeHash(parms.entryType === undefined ? ANCHOR_GENERIC : parms.entryType, makeAnchorValue(parms.entryType === undefined ? ANCHOR_GENERIC : parms.entryType, parms.value));
-}
-
-// anchor objects always contain their entryType so as to generate unique hashes across the same value with a different entryType
-function makeAnchorValue(entryType, value)
-{
-    return value;
-}
-
-// decodes an anchor object stored on the chain back to it's value prior to the entryType addition that happened in makeAnchorObject
-function getAnchorValue(obj)
-{
-    return obj;
-}
-
-function isObject(item)
-{
-    return item === Object(item);
+    return false;
```

Comment:
condense into one line of code:
return entry_type === 'anchor' || entry_type === 'anchor_link';